src/LanguageServer/Protocol/DefaultCapabilitiesProvider.cs

```diff
@@ -90,10 +90,13 @@ public ServerCapabilities GetCapabilities(ClientCapabilities clientCapabilities)
             // Using only range handling has shown to be more performant than using a combination of full/edits/range
             // handling, especially for larger files. With range handling, we only need to compute tokens for whatever
             // is in view, while with full/edits handling we need to compute tokens for the entire file and then
-            // potentially run a diff between the old and new tokens.
+            // potentially run a diff between the old and new tokens. Therefore, we only enable full handling if
+            // the client does not support ranges.
+            var rangeCapabilities = clientCapabilities.TextDocument?.SemanticTokens?.Requests?.Range;
+            var supportsSemanticTokensRange = rangeCapabilities?.Value is not (false or null);
```

Comment:
is this just asking `is true`?

Reply:
Range here is a sum type that can be either `bool` or `object?` so if you just did `is true` it would return false if there is an object instead, in which case the client would actually have the range capability, afaik